tls/s2n_signature_algorithms.c

```diff
@@ -88,38 +112,22 @@ int s2n_choose_sig_scheme(const struct s2n_signature_scheme* const* our_pref_lis
     S2N_ERROR(S2N_ERR_INVALID_SIGNATURE_ALGORITHM);
 }
 
-int s2n_get_signature_scheme_pref_list(struct s2n_connection *conn,
-                                       const struct s2n_signature_scheme* const** pref_list_out, size_t *list_len_out) {
-
-    /* Our SignatureScheme Preference list depends on the TLS version that was negotiated */
-
-    const struct s2n_signature_scheme* const* our_pref_list = s2n_legacy_sig_scheme_pref_list;
-    size_t our_pref_len =  s2n_legacy_sig_scheme_pref_list_len;
-
-    if (conn->actual_protocol_version == S2N_TLS13) {
-        our_pref_list = s2n_tls13_sig_scheme_pref_list;
-        our_pref_len = s2n_tls13_sig_scheme_pref_list_len;
-    }
-
-    *pref_list_out = our_pref_list;
-    *list_len_out = our_pref_len;
-
-    return 0;
-}
-
 
 int s2n_get_and_validate_negotiated_signature_scheme(struct s2n_connection *conn, struct s2n_stuffer *in,
                                              struct s2n_signature_scheme *chosen_sig_scheme) {
     uint16_t actual_iana_val;
     GUARD(s2n_stuffer_read_uint16(in, &actual_iana_val));
 
-    const struct s2n_signature_scheme* const* our_pref_list;
-    size_t our_pref_len;
+    const struct s2n_signature_preferences *signature_preferences = conn->config->signature_preferences;
+    notnull_check(signature_preferences);
 
-    GUARD(s2n_get_signature_scheme_pref_list(conn, &our_pref_list, &our_pref_len));
+    for (int i = 0; i < signature_preferences->count; i++) {
+        const struct s2n_signature_scheme *candidate = signature_preferences->signature_schemes[i];
+
+        if (0 != s2n_signature_scheme_valid_to_accept(conn, candidate)) {
```

Comment:
nitpick: I think our convention is to put the numeric on the RHS of the comparison